lib/synapse/haproxy.rb

```diff
@@ -670,6 +688,17 @@ def generate_frontend_stanza(watcher, config)
     end
 
     def generate_backend_stanza(watcher, config)
+      backends = {}
+
+      @seen.fetch(watcher.name, []).each do |backend_name, backend|
+        backends[backend_name] = backend.merge('enabled' => false)
+      end
+
+      watcher.backends.each do |backend|
```

Comment:
Can you add a comment that the ordering here is important so that we always enable alive backends?